client/data/onboarding/selectors.ts

```diff
@@ -14,5 +17,36 @@ export const getOnboardingState = ( state: State ): OnboardingState => {
 };
 
 export const getBusinessTypes = ( state: State ): Country[] => {
-	return getOnboardingState( state ).data || [];
+	const businessTypes = getOnboardingState( state ).data || [];
+
+	return businessTypes.map( ( country ) => {
+		const types = country.types.map( ( type ) => {
+			const structures = type.structures.map( ( structure ) => {
+				return {
+					key: structure.key,
+					name:
+						businessStructureStrings[ country.key ][
+							structure.key
+						],
+				};
+			} );
+
+			return {
+				key: type.key,
+				name: businessTypeStrings[ type.key ],
+				description: businessTypeDescriptionStrings.hasOwnProperty(
+					country.key
+				)
+					? businessTypeDescriptionStrings[ country.key ][ type.key ]
+					: businessTypeDescriptionStrings.generic[ type.key ],
+				structures: structures,
+			};
+		} );
+
+		return {
+			key: country.key,
+			name: country.name,
+			types: types,
+		};
+	} );
```

Comment:
Nothing wrong here, but we can write it a bit more compact and easy to read:
- Getting rid of `return` in a close that returns an object.
- Checking if an object has a property accessing it directly rather than calling `hasOwnProperty`.

```js
x.map( ( y ) => ( { z: y } ) );
// Both are the same
x.map( ( y ) => {
  return {
    z: y,
  };
} );
```

Anyway, feel free to leave it as is, if you find the actual code more readable 🙂

```suggestion
	return businessTypes.map( ( country ) => ( {
		key: country.key,
		name: country.name,
		types: country.types.map( ( type ) => ( {
			key: type.key,
			name: businessTypeStrings[ type.key ],
			description: businessTypeDescriptionStrings[ country.key ]
				? businessTypeDescriptionStrings[ country.key ][ type.key ]
				: businessTypeDescriptionStrings.generic[ type.key ],
			structures: type.structures.map( ( structure ) => ( {
				key: structure.key,
				name: businessStructureStrings[ country.key ][ structure.key ],
			} ) ),
		} ) ),
	} ) );
```

Reply:
Thanks! I like your suggestion and I committed it. I didn't know you could directly return the object in the callback of the mapping function, and I'm still in the habit of doing `hasOwnProperty` from the old days of JS 😅